### PR TITLE
[Keyboard] Add OSA keyboard

### PR DESCRIPTION
**Pull request by jrfhoutx**
## Description

Adding support for the OSA keyboard 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sck/osa/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+![osa](imgur.com image replace me!)
+
+About the OSA:
+
+The OSA was run as a private group buy of 20 units. It is an Alice layout keyboard based on the Open Source Alice files, altered by BlindAssassin111 based on input from the private group buy members.
+All case and other design files have been either reworked or created from scratch by BlindAssassin111. PCB was designed from scratch by BlindAssassin111.
+The OSA PCB is an ATMega32u4 powered, USB Type C, Alice compatible PCB and was created specifically for the OSA case, but with considerations to make it compatible with the Lubrigante acrylic cases (except for the Type C connector it should also be compatible with original TGR Alice cases).
+
+Keyboard Maintainer: [jrfhoutx](https://github.com/jrfhoutx)  
+Hardware Supported: OSA PCB by BlindAssassin111, ATMEGA32u4
```
Comment (mechmerlin) [inline keyboards/sck/osa/readme.md]: Need extra space for markdown to format correctly

```suggestion
Hardware Supported: OSA PCB by BlindAssassin111, ATMEGA32u4  
```

Comment (mechmerlin) [inline keyboards/sck/osa/readme.md]: Need extra space for markdown to render correctly

```suggestion
Hardware Supported: OSA PCB by BlindAssassin111, ATMEGA32u4  
```

File: keyboards/sck/osa/readme.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# osa
+
+![osa](imgur.com image replace me!)
```
Comment (mechmerlin) [inline keyboards/sck/osa/readme.md]: Remove this if no image is being showed

File: keyboards/sck/osa/osa.h [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   KC_NO  }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   KC_NO  }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   KC_NO  }, \
+    { KC_NO, K41,   KC_NO, K43,   KC_NO, K45,   K46,   KC_NO  }, \
+    { K50,   K51,   K52,   K53,   K54,   K55,   K56,   K57    }, \
+    { K60,   K61,   K62,   K63,   K64,   K65,   K66,   K67    }, \
+    { KC_NO, K71,   K72,   K73,   K74,   K75,   K76,   K77    }, \
+    { K80,   K81,   K82,   K83,   K84,   K85,   K86,   K87    }, \
+    { K90,   KC_NO, KC_NO, K93,   KC_NO, K95,   KC_NO, KC_NO  }  \
+}
```
Comment (noroadsleft) [inline keyboards/sck/osa/osa.h]: The keyboard's header file needs to include the layout macros that `info.json` is referencing.

```suggestion
}

#define LAYOUT_default_ansi( \
  K10, K00, K01, K02, K03, K04, K05, K06,   K56, K55, K54, K53, K52, K51,      K57, \
  K20, K11, K12, K13, K14, K15, K16,        K66, K65, K64, K63, K62, K61, K60, K67, \
  K30, K21, K22, K23, K24, K25, K26,        K76, K75, K74, K73, K72, K71,      K77, \
       K31, K32, K33, K34, K35, K36,        K86, K85, K84, K83, K82, K81,      K87, \
       K41,      K43,      K45, K46,        K95,           K93,                K90  \
) \
{ \
    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   KC_NO  }, \
    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   KC_NO  }, \
    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   KC_NO  }, \
    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   KC_NO  }, \
    { KC_NO, K41,   KC_NO, K43,   KC_NO, K45,   K46,   KC_NO  }, \
    { KC_NO, K51,   K52,   K53,   K54,   K55,   K56,   K57    }, \
    { K60,   K61,   K62,   K63,   K64,   K65,   K66,   K67    }, \
    { KC_NO, K71,   K72,   K73,   K74,   K75,   K76,   K77    }, \
    { KC_NO, K81,   K82,   K83,   K84,   K85,   K86,   K87    }, \
    { K90,   KC_NO, KC_NO, K93,   KC_NO, K95,   KC_NO, KC_NO  }  \
}

#define LAYOUT_split_backspace( \
  K10, K00, K01, K02, K03, K04, K05, K06,   K56, K55, K54, K53, K52, K51, K50, K57, \
  K20, K11, K12, K13, K14, K15, K16,        K66, K65, K64, K63, K62, K61, K60, K67, \
  K30, K21, K22, K23, K24, K25, K26,        K76, K75, K74, K73, K72, K71,      K77, \
       K31, K32, K33, K34, K35, K36,        K86, K85, K84, K83, K82, K81,      K87, \
       K41,      K43,      K45, K46,        K95,           K93,                K90  \
) \
{ \
    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   KC_NO  }, \
    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   KC_NO  }, \
    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   KC_NO  }, \
    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   KC_NO  }, \
    { KC_NO, K41,   KC_NO, K43,   KC_NO, K45,   K46,   KC_NO  }, \
    { K50,   K51,   K52,   K53,   K54,   K55,   K56,   K57    }, \
    { K60,   K61,   K62,   K63,   K64,   K65,   K66,   K67    }, \
    { KC_NO, K71,   K72,   K73,   K74,   K75,   K76,   K77    }, \
    { KC_NO, K81,   K82,   K83,   K84,   K85,   K86,   K87    }, \
    { K90,   KC_NO, KC_NO, K93,   KC_NO, K95,   KC_NO, KC_NO  }  \
}

#define LAYOUT_split_right_shift( \
  K10, K00, K01, K02, K03, K04, K05, K06,   K56, K55, K54, K53, K52, K51,      K57, \
  K20, K11, K12, K13, K14, K15, K16,        K66, K65, K64, K63, K62, K61, K60, K67, \
  K30, K21, K22, K23, K24, K25, K26,        K76, K75, K74, K73, K72, K71,      K77, \
       K31, K32, K33, K34, K35, K36,        K86, K85, K84, K83, K82, K81, K80, K87, \
       K41,      K43,      K45, K46,        K95,           K93,                K90  \
) \
{ \
    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   KC_NO  }, \
    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   KC_NO  }, \
    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   KC_NO  }, \
    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   KC_NO  }, \
    { KC_NO, K41,   KC_NO, K43,   KC_NO, K45,   K46,   KC_NO  }, \
    { KC_NO, K51,   K52,   K53,   K54,   K55,   K56,   K57    }, \
    { K60,   K61,   K62,   K63,   K64,   K65,   K66,   K67    }, \
    { KC_NO, K71,   K72,   K73,   K74,   K75,   K76,   K77    }, \
    { K80,   K81,   K82,   K83,   K84,   K85,   K86,   K87    }, \
    { K90,   KC_NO, KC_NO, K93,   KC_NO, K95,   KC_NO, KC_NO  }  \
}
```

File: keyboards/sck/osa/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+   * | M2 ||Tab  |  Q|  W|  E|  R|  T|          |  Y|  U|  I|  O|  P| [ | ] |  \  |
+   * |----||----------------------------------------------------------------------|
+   * | M3 ||Ctrl   |  A|  S|  D|  F|  G|          |  H|  J|  K|  L|  ;|  '|Return |
+   * `----'|----------------------------------------------------------------------|
+   *       |Shift   |  Z|  X|  C|  V|  B|     |  B|  N|  M|  ,|  .|  /|Shift      |
+   *       |----------------------------------------------------------------------|
+   *       |CAPS|    | Alt | Space | Fn |     | Space | Alt |               | Ctrl|
+   *       `----------------------------------------------------------------------'
+   */
+  [0] = LAYOUT( /* Base */
```
Comment (noroadsleft) [inline keyboards/sck/osa/keymaps/default/keymap.c]:
```suggestion
  [0] = LAYOUT_all( /* Base */
```

File: keyboards/sck/osa/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+   * |RGB_MOD ||Tab  |  Q|  W|  E|  R|  T|          |  Y|  U|  I|  O|  P| [ | ] |  \  |
+   * |--------||----------------------------------------------------------------------|
+   * |RGB_RMOD||Ctrl   |  A|  S|  D|  F|  G|          |  H|  J|  K|  L|  ;|  '|Return |
+   * `--------'|----------------------------------------------------------------------|
+   *           |Shift   |  Z|  X|  C|  V|  B|     |  B|  N|  M|  ,|  .|  /|Shift      |
+   *           |----------------------------------------------------------------------|
+   *           |CAPS|    | Alt | Space | Fn |     | Space | Alt |               | Ctrl|
+   *           `----------------------------------------------------------------------'
+   */
+    [1] = LAYOUT( /* OSA Layer 1 */
```
Comment (noroadsleft) [inline keyboards/sck/osa/keymaps/default/keymap.c]:
```suggestion
    [1] = LAYOUT_all( /* OSA Layer 1 */
```

**Comment by jrfhoutx** [comment]
Not all of the commented out code has been removed (like RGB code) as some of it is left for users to be able to easily comment in/out for their own use. 

Other than that, all other issues raised should be addressed and fixed, and all requested changes made.

File: keyboards/sck/osa/rules.mk [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
```
Comment (fauxpark) [inline keyboards/sck/osa/rules.mk]:
```suggestion
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

File: keyboards/sck/osa/osa.h (hunk 3 above)
Comment (jrfhoutx) [reply keyboards/sck/osa/osa.h]: This should be correct now.

File: keyboards/sck/osa/info.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (noroadsleft) [inline keyboards/sck/osa/info.json]: This file should be reverted to the version in b3b49c3700151ab8613a50907aa819492099d444.

```
git checkout b3b49c3 -- keyboards/sck/osa/info.json
git commit
```

File: keyboards/sck/osa/readme.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
+# osa
+
+![osa](https://i.imgur.com/U2x7TQ8)
```
Comment (noroadsleft) [inline keyboards/sck/osa/readme.md]: Needs the file extension present for the image to show.

```suggestion
![osa](https://i.imgur.com/U2x7TQ8.jpg)
```

File: keyboards/sck/osa/info.json (hunk 7 above)
Comment (jrfhoutx) [reply keyboards/sck/osa/info.json]: reverted as requested

For my future reference, is configurator not capable of rendering/using the angled alphas?

Comment (noroadsleft) [reply keyboards/sck/osa/info.json]: Configurator does not currently support rotated keys. You *could* set up the file so the keys are unrotated but staggered vertically, like the Atreus62 or Ergodox EZ do, but that requires editing the `info.json` by hand for basically every key on every layout. I didn't feel like putting you through that.

Comment (jrfhoutx) [reply keyboards/sck/osa/info.json]: ok thank you

**Comment by jrfhoutx** [comment]
Thanks!

**Pull request merged**